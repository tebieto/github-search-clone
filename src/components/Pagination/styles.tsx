import styled from 'styled-components';
import colors from '../../utils/colors';

export const PaginationContainer = styled.div`
  margin: 50px 0;
  display: flex;
  justify-content: flex-end;
  align-items: center;
  @media screen and (max-width: 800px) {
    justify-content: center;
  }
  .pages,
  .prefix,
  .suffix {
    display: flex;
    justify-content: center;
    .page {
      color: ${colors.secondaryTextColor};
      margin: 0 12px;
      font-size: 18px;
      cursor: pointer;
    }
    .active {
      color: ${colors.primaryTextColor};
    }
  }

  .icon {
    background-color: ${colors.primaryHoverColor};
    color: ${colors.tertiaryTextColor};
    font-size: 35px;
    padding: 5px;
    border-radius: 10px;
    cursor: pointer;
  }

  .active-icon {
    background-color: ${colors.primaryColor};
    color: ${colors.secondaryColor};
  }

  .arrow-left {
    margin-right: 10px;
  }

  .arrow-right {
    margin-left: 10px;
  }
`;
