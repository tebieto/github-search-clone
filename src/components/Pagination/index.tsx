import React from 'react';
import { PaginationContainer } from './styles';
import KeyboardArrowLeftIcon from '@material-ui/icons/KeyboardArrowLeft';
import KeyboardArrowRightIcon from '@material-ui/icons/KeyboardArrowRight';
import { getPageButtons } from '../../utils';
interface PaginationProps {
  itemsLength: number;
  currentPage: number;
  maxItemPerPage: number;
  setCurrentPage: { (page: number): void };
}
const Pagination = ({
  itemsLength,
  maxItemPerPage,
  currentPage,
  setCurrentPage,
}: PaginationProps): JSX.Element | null => {
  const maxButtonLength = 4;
  const buttonLength = Math.ceil(itemsLength / maxItemPerPage);
  const pageButtons = getPageButtons({
    buttonLength,
    maxButtonLength,
    currentPage,
  });
  const activatePrevious = currentPage > 0;
  const activateNext = currentPage < buttonLength - 1;
  return itemsLength ? (
    <PaginationContainer>
      <KeyboardArrowLeftIcon
        onClick={() => activatePrevious && setCurrentPage(currentPage - 1)}
        className={`icon arrow-left ${activatePrevious && 'active-icon'}`}
      />
      <div className="pages">
        {currentPage > 0 + maxButtonLength - 1 ? (
          <div className="prefix">
            <div onClick={() => setCurrentPage(0)} className="page">
              {1}
            </div>
            <div className="page">...</div>
          </div>
        ) : null}
        {pageButtons.map((page: number) => {
          return (
            <div
              onClick={() => setCurrentPage(page)}
              className={`page ${page === currentPage && 'active'}`}
              key={page}
            >
              {page + 1}
            </div>
          );
        })}
        {pageButtons[pageButtons.length - 1] !== buttonLength - 1 ? (
          <div className="suffix">
            <div className="page">...</div>
            <div
              onClick={() => setCurrentPage(buttonLength - 1)}
              className="page"
            >
              {buttonLength}
            </div>
          </div>
        ) : null}
      </div>
      <KeyboardArrowRightIcon
        onClick={() => activateNext && setCurrentPage(currentPage + 1)}
        className={`icon arrow-right ${activateNext && 'active-icon'}`}
      />
    </PaginationContainer>
  ) : null;
};

export default Pagination;
