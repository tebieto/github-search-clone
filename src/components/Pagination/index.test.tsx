import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Pagination from '.';

it('renders correctly', () => {
  const tree = shallow(
    <Pagination
      itemsLength={100}
      setCurrentPage={() => null}
      currentPage={0}
      maxItemPerPage={10}
    />,
  );
  expect(toJson(tree)).toMatchSnapshot();
});
