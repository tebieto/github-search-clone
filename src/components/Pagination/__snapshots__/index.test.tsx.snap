// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <Memo(ForwardRef(KeyboardArrowLeftIcon))
    className="icon arrow-left false"
    onClick={[Function]}
  />
  <div
    className="pages"
  >
    <div
      className="page active"
      key="0"
      onClick={[Function]}
    >
      1
    </div>
    <div
      className="page false"
      key="1"
      onClick={[Function]}
    >
      2
    </div>
    <div
      className="page false"
      key="2"
      onClick={[Function]}
    >
      3
    </div>
    <div
      className="page false"
      key="3"
      onClick={[Function]}
    >
      4
    </div>
    <div
      className="suffix"
    >
      <div
        className="page"
      >
        ...
      </div>
      <div
        className="page"
        onClick={[Function]}
      >
        10
      </div>
    </div>
  </div>
  <Memo(ForwardRef(KeyboardArrowRightIcon))
    className="icon arrow-right active-icon"
    onClick={[Function]}
  />
</styled.div>
`;
